WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Reprint Shipping Label/ReprintShippingLabelCoordinator.swift

```diff
@@ -0,0 +1,112 @@
+import UIKit
+import struct Yosemite.ShippingLabel
+import struct Yosemite.ShippingLabelPrintData
+import enum Yosemite.ShippingLabelPaperSize
+
+/// Coordinates navigation for reprinting a shipping label.
+protocol ReprintShippingLabelCoordinatorProtocol: class {
+    /// Shows the main reprint screen given a shipping label.
+    /// - Parameter shippingLabel: The shipping label to reprint.
+    func showReprintUI(shippingLabel: ShippingLabel)
+
+    /// Allows the user to select a paper size from supported options.
+    /// - Parameters:
+    ///   - paperSizeOptions: All paper sizes that we support for reprinting.
+    ///   - selectedPaperSize: The currently selected paper size.
+    ///   - onPaperSizeSelected: Called when the user leaves the paper size selector with the selected value.
+    func showPaperSizeSelector(paperSizeOptions: [ShippingLabelPaperSize],
+                               selectedPaperSize: ShippingLabelPaperSize?,
+                               onPaperSizeSelected: @escaping (ShippingLabelPaperSize?) -> Void)
+
+    /// Presents an in-progress modal when requesting shipping label document for reprinting.
+    func presentReprintInProgressUI()
+
+    /// Dismisses the in-progress modal when the shipping label document result is ready.
+    /// When the result is successful, we present an AirPrint modal.
+    /// If the result fails with an error, we present an error alert.
+    /// - Parameter result: The result of shipping label document request, could be `ShippingLabelPrintData` on success or an error.
+    func dismissReprintInProgressUIAndPresentPrintingResult(_ result: Result<ShippingLabelPrintData, ReprintShippingLabelError>)
+}
+
+/// Implements `ReprintShippingLabelCoordinatorProtocol` that handles all navigation actions.
+final class ReprintShippingLabelCoordinator {
+    private let sourceViewController: UIViewController
```

Comment:
oops great catch, my bad! I was hoping to make the coordinator have the same life cycle as reprint UX to avoid the retain cycle but hadn't figured out a way. Also forgot to double check with memory debugger 😅